src/main/java/Bot/DateConverter.java

```diff
@@ -0,0 +1,115 @@
+package Bot;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ Responsible for providing methods for formatting user input date string into readable date string format
+ */
+public class DateConverter {
+
+    /**
+     * Converts the given date string to a readable format. It determines if the
+     * input is in a datetime format or a date-only format and applies the appropriate conversion.
+     *
+     * @param input The date string to be converted.
+     * @return The formatted date string or an error message if the format is invalid.
+     */
+    public static String convertDate(String input) {
```

Comment:
would it be better to name it as convertDateToString so that it becomes more specific?